audacity/audacity
# 7369 au4 tone generator

Resolves: #7369
May resolve #7509 

*(short description of the changes and the motivation to make the changes)*




Recommended:

### QA:
Note that the tone generator is still to be found under **Effects > Built-in > Tone**. This must be addressed in a follow-up PR.

The UI is still unbalanced ; for now, please ignore as long as the functionality is here.

An effort was made to make it on-par with AU3's tone generator. Please use your AU3 tone-generator regression test plan and report missing bits.

Some things to make sure of (not exhaustive):
- [x] Generate on empty project creates new track
- [x] When no track is selected, generating also creates a new track.
- [x] Generate over existing clip replaces selection with tone.
- [x] If there isn't a time selection when invoking the effect, the duration format is `hh::mm::ss + milliseconds`.
- [x] If there is a time selection when invoking the effect, the duration format is `hh:mm:ss + samples`,
- [x] Generating when there is not enough room shows the expected error dialog,
